### PR TITLE
Added synced net messenger

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- added syncedNetworkMessenger
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/processingOnlyNode/syncedMessenger.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+}
+
+// UnregisterMessageProcessor will unregister the message processor for the provided topic and identifier
+func (messenger *syncedMessenger) UnregisterMessageProcessor(topic string, identifier string) error {
+	messenger.mutOperation.Lock()
+	defer messenger.mutOperation.Unlock()
+
+	handlers, found := messenger.topics[topic]
+	if !found {
+		return fmt.Errorf("programming error in syncedMessenger.UnregisterMessageProcessor, topic %s does not exists", topic)
```
Comment (bogdan-rosianu) [inline node/processingOnlyNode/syncedMessenger.go:157]: does not exist

File: node/processingOnlyNode/syncedMessenger.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+}
+
+// Verify will check if the provided signature === hash(pid + payload)
+func (messenger *syncedMessenger) Verify(payload []byte, pid core.PeerID, signature []byte) error {
+	sig := hasher.Compute(pid.Pretty() + string(payload))
+	if bytes.Equal(sig, signature) {
+		return nil
+	}
+
+	return errors.New("invalid signature")
```
Comment (bogdan-rosianu) [inline node/processingOnlyNode/syncedMessenger.go:333]: might define a variable for this error at the top of the page (`var` section)

File: node/processingOnlyNode/syncedMessenger.go (hunk 1 above)
Comment (iulianpascalau) [reply node/processingOnlyNode/syncedMessenger.go:157]: refactored

File: node/processingOnlyNode/syncedMessenger.go (hunk 2 above)
Comment (iulianpascalau) [reply node/processingOnlyNode/syncedMessenger.go:333]: defined for all errors

File: node/processingOnlyNode/syncedBroadcastNetwork_test.go [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+	assert.Contains(t, peers, peer1.ID())
+	assert.Contains(t, peers, peer2.ID())
+
+	assert.True(t, peer1.IsConnected(peer2.ID()))
+	assert.True(t, peer2.IsConnected(peer1.ID()))
+	assert.False(t, peer1.IsConnected("no connection"))
+
+	addresses := peer1.ConnectedAddresses()
+	assert.Equal(t, 2, len(addresses))
+	fmt.Println(addresses)
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/processingOnlyNode/syncedBroadcastNetwork_test.go:232]: done

**Review by miiu96 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**